src/app.tsx

```diff
@@ -100,8 +162,65 @@ function App() {
       <Button variant="dark" onClick={dummy}>
         Hello
       </Button>
+
+      <div className="flashcards-container">
+        <Flashcard card={readonlyCard} variant={'readonly'} />
+        <Reorder.Group axis="y" values={cards} onReorder={setCards}>
+          {getFlashcards()}
+        </Reorder.Group>
+      </div>
     </div>
   );
+
+  function getFlashcards() {
+    return cards.map((card, id) => {
+      const index = id + 1;
+      const value = (
+        <Flashcard
+          key={index}
+          card={card}
+          index={index}
+          variant={activeCard === card.id ? 'active' : 'inactive'}
+          onFocus={() => setActiveCard(card.id)}
+          onCardChange={(c) => handleCardChange(c, id)}
+          onDownClick={() => handleDownClick(id)}
+          onUpClick={() => handleUpClick(id)}
+          onRemoveClick={() => handleRemoveClick(id)}
+        />
+      );
+      return (
+        <Reorder.Item key={card.id} value={card}>
+          {value}
+        </Reorder.Item>
+      );
+    });
+
+    function handleRemoveClick(id: number) {
+      const newCards = [...cards];
+      newCards.splice(id, 1);
+      setCards(newCards);
+    }
+
+    function handleUpClick(id: number) {
+      // Todo handle up click when card is first
```

Comment:
This actually works already, negative indices in `splice` define the position from the end of the array.

Reply:
huh, good to know, haha.